src/ucp/core/ucp_context.c

```diff
@@ -125,6 +125,11 @@ static ucs_config_field_t ucp_config_table[] = {
    "MD whose distance is queried when evaluating transport selection score",
    ucs_offsetof(ucp_config_t, selection_cmp), UCS_CONFIG_TYPE_STRING},
 
+  {"MEMTYPE_REG_WHOLE_ALLOC_TYPES", "",
+   "Memory types which have whole allocations registered.\n"
+   "Allowed memory types: cuda, cuda-managed, rocm, rocm-managed",
+   ucs_offsetof(ucp_config_t, ctx.reg_whole_alloc), UCS_CONFIG_TYPE_STRING_ARRAY},
```

Comment:
we dont do rdma for managed right? do we need those in this list

Reply:
@bureddy This doesn't force cuda-managed to be registered with IB. It would mean that if IB were allowed to register cuda-managed buffers, it would register whole allocations as opposed to just user passed range.

Reply:
currently, we disable rdma with managed memory and none of the MDs can register it.  so there is no effect in allowing them in this list.